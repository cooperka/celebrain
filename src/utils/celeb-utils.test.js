import imageData from '../celebs.json';

import celebUtils from './celeb-utils';

describe('celeb utils', () => {
  it('gets display names for celebs.json', () => {
    const displayName = celebUtils.getDisplayName(imageData[0]);
    expect(displayName).toMatchSnapshot();
  });

  const titles = ['', 'Foo', 'Foo Bar', 'Foo Bar Baz', 'Foo Bar (Actor)'];

  titles.forEach((title) => {
    it(`gets display full-name for '${title}'`, () => {
      const displayName = celebUtils.getDisplayName({ title });
      expect(displayName).toMatchSnapshot();
    });
  });

  titles.forEach((title) => {
    it(`gets display first-name for '${title}'`, () => {
      const displayName = celebUtils.getDisplayName({ title }, true);
      expect(displayName).toMatchSnapshot();
    });
  });
});
