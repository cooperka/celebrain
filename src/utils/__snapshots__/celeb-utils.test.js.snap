// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`celeb utils gets display first-name for '' 1`] = `""`;

exports[`celeb utils gets display first-name for 'Foo Bar (Actor)' 1`] = `"Foo"`;

exports[`celeb utils gets display first-name for 'Foo Bar Baz' 1`] = `"Foo"`;

exports[`celeb utils gets display first-name for 'Foo Bar' 1`] = `"Foo"`;

exports[`celeb utils gets display first-name for 'Foo' 1`] = `"Foo"`;

exports[`celeb utils gets display full-name for '' 1`] = `""`;

exports[`celeb utils gets display full-name for 'Foo Bar (Actor)' 1`] = `"Foo Bar"`;

exports[`celeb utils gets display full-name for 'Foo Bar Baz' 1`] = `"Foo Bar Baz"`;

exports[`celeb utils gets display full-name for 'Foo Bar' 1`] = `"Foo Bar"`;

exports[`celeb utils gets display full-name for 'Foo' 1`] = `"Foo"`;

exports[`celeb utils gets display names for celebs.json 1`] = `"Arnold Schwarzenegger"`;
